inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.cpp

```diff
@@ -334,94 +361,275 @@ bool MKLDNNDeconvolutionNode::created() const {
+            wgh_candidate = internalBlobMemory.front()->GetDescWithType<DnnlMemoryDesc>()->getDnnlDesc();
+        }
+    } else {
+        wgh_candidate = getParentEdgesAtPort(1).front()->getMemory().GetDescWithType<DnnlMemoryDesc>()->getDnnlDesc();
+    }
+
+    std::shared_ptr<MKLDNNDescriptor> desc;
+    if (isInt8) {
+        desc = std::make_shared<MKLDNNDescriptor>(createDescriptorInternalInt8(in_candidate, wgh_candidate, out_candidate,
+                                                  mkldnn::algorithm::deconvolution_direct));
+    } else {
+        mkldnn::algorithm alg = (selected_pd->getImplementationType() == MKLDNNPlugin::impl_desc_type::jit_avx512_winograd) ?
+                                 mkldnn::algorithm::convolution_winograd : mkldnn::algorithm::convolution_direct;
+        std::shared_ptr<convolution_backward_data::desc> deconv_desc;
+        std::shared_ptr<convolution_forward::primitive_desc> fwd_conv_pd;
+        std::tie(deconv_desc, fwd_conv_pd) = createDescriptorInternalDefault(in_candidate, wgh_candidate, out_candidate, alg);
+        if (fwd_conv_pd->get(true) == nullptr) {
+            IE_THROW() << "Forward convolution primitive descriptor is nullable for node with name: " << getName();
+        }
+        desc = std::make_shared<MKLDNNDescriptor>(deconv_desc, fwd_conv_pd);
+    }
+
+    auto itpd = desc->createPrimitiveDescriptorIterator(getEngine(), *pAttrLocal);
+
+    const auto &src = getParentEdgesAtPort(0)[0]->getMemoryPtr()->GetPrimitive();
+    const auto &dst = getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetPrimitive();
+
+    while (static_cast<bool>(itpd)) {
+        impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
 
-        prim.reset(new convolution_backward_data(prim_desc));
+        if (isInt8) {
+            if (impl_type == selected_pd->getImplementationType() ||
+                    // WA: when we have 'output_shape' as parametr we choose selected primitive descriptor as jit
+                    // but for some 'output_shape' values we can execute only on ref implementation
+                    ((selected_pd->getImplementationType() & jit) && (impl_type == ref_any))) {
+                if (internalBlobMemory.empty()) {
+                    prepareMemory(selected_pd, itpd);
+                }
+                auto prim_desc = deconvolution_forward::primitive_desc(itpd.get());
+                prim.reset(new deconvolution_forward(prim_desc));
+                primArgs = {{DNNL_ARG_SRC, src}, {DNNL_ARG_WEIGHTS, getWeights()}, {DNNL_ARG_DST, dst}};
+                break;
+            }
+        } else {
+            if (impl_type == selected_pd->getImplementationType() ||
+                    // WA: when we have 'output_shape' as parametr we choose selected primitive descriptor as jit
+                    // but for some 'output_shape' values we can execute only on ref implementation
+                    ((selected_pd->getImplementationType() & jit) && (impl_type == ref_any))) {
```

Comment:
In such scenario we expected using gemm implementation with additional internal reorders. Is this approach applicable in this circumstances?